content/blog/using-apache-spark-dataframes-for-processing-of-tabular-data.md

```diff
@@ -0,0 +1,362 @@
+---
+title: Using Apache Spark DataFrames for Processing of Tabular Data
+date: 2021-10-14T05:35:43.497Z
+priority: 4
+author: Carol McDonald
+authorimage: /img/Avatar1.svg
+tags:
+  - HPE Ezmeral Data Fabric
+  - MapR
+  - Apache Spark
+---
+```
+"authorDisplayName": "Carol McDonald",
+"category": "apache-spark",
+"publish": "2015-06-24T07:00:00.000Z"
+```
+
+**Editor’s Note:** MapR products and solutions sold prior to the acquisition of such assets by Hewlett Packard Enterprise Company in 2019 may have older product names and model numbers that differ from current solutions. For information about current offerings, which are now part of HPE Ezmeral Data Fabric, please visit <https://www.hpe.com/us/en/software/ezmeral-data-fabric.html>
+
+This post will help you get started using Apache Spark DataFrames with Scala on the MapR Sandbox. The <a target='\_blank'  href='http://spark.apache.org/docs/latest/sql-programming-guide.html#datasets-and-dataframes'>new Spark DataFrames API</a> is designed to make big data processing on tabular data easier.
+
+## What is a Spark DataFrame?
+
+A Spark DataFrame is a distributed collection of data organized into named columns that provides operations to filter, group, or compute aggregates, and can be used with Spark SQL. DataFrames can be constructed from structured data files, existing RDDs, tables in Hive, or external databases.
+
+In this post, you’ll learn how to:
+
+* Load data into Spark DataFrames
+* Explore data with Spark SQL
+
+This post assumes a basic understanding of Spark concepts. If you have not already read the tutorial on [Getting Started with Spark on MapR Sandbox](https://developer.hpe.com/blog/using-apache-spark-dataframes-for-processing-of-tabular-data-1/), it would be good to read that first.
+
+## Software
+
+This tutorial will run on the MapR v5.0 Sandbox, which includes Spark 1.3
+
+* You can download the code and data to run these examples from here:
+
+  * <a target='\_blank'  href='https://github.com/caroljmcdonald/sparkdataframeexample'>https://github.com/caroljmcdonald/sparkdataframeexample</a>
+* The examples in this post can be run in the spark-shell, after launching with the spark-shell command.
+* You can also run the code as a standalone application as described in the tutorial on [Getting Started with Spark on MapR Sandbox](https://developer.hpe.com/blog/using-apache-spark-dataframes-for-processing-of-tabular-data-1/).
+
+**The sample data sets**
+
+We will use two example datasets - one from <a target='\_blank'  href='http://www.modelingonlineauctions.com/datasets'>eBay online auctions</a> and one from the <a target='\_blank'  href='https://data.sfgov.org/Public-Safety/SFPD-Incidents-from-1-January-2003/tmnf-yvry'>SFPD Crime Incident Reporting system</a>.
+
+The eBay online auction dataset has the following data fields:
+
+***auctionid*** - unique identifier of an auction\
```

Comment:
Remove the \ at the end of each item element in the list

Reply:
done. That slash was previously used to force a carriage return. I replaced it with two spaces